### PR TITLE
[gitlab] inherit generic e2e rule for e2e_pre_test

### PR DESCRIPTION
**Pull request by pducolin**
### What does this PR do?

Inherit `e2e_pre_test` rules from generic e2e rules

### Motivation

This job is currently running on child pipelines triggered by `integrations-core`. Today this caused an incident as `integrations-core` triggered 50 pipelines on the same commit, which had us reach the vCPU limit on the `datadog-agent-qa` AWS account, tracked by incident-28175

### CODE REVIEW

**Review by KevinFairise2 — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -814,6 +796,20 @@ workflow:
         - test/new-e2e/go.mod
       compare_to: main # TODO: use a variable, when this is supported https://gitlab.com/gitlab-org/gitlab/-/issues/369916
 
+.on_e2e_or_fakeintake_changes_or_manual:
+  - !reference [.on_e2e_main_release_or_rc]
```
[the remark this answers is not in the document]
Comment (pducolin) [reply .gitlab-ci.yml:800]: `if_disable_e2e_tests` expands to

```yaml
.if_disable_e2e_tests: &if_disable_e2e_tests
  if: $RUN_E2E_TESTS == "off"
```

https://github.com/DataDog/datadog-agent/blob/pducolin/incident-28175-do-not-run-e2e-pre-test-on-child-pipelines/.gitlab-ci.yml#L332-L333

[the remark this answers is not in the document]
Comment (pducolin) [reply .gitlab-ci.yml:800]: `if_run_all_e2e_tests` expands to

```yaml
.if_run_all_e2e_tests: &if_run_all_e2e_tests
  if: $RUN_E2E_TESTS == "on"
```

https://github.com/DataDog/datadog-agent/blob/pducolin/incident-28175-do-not-run-e2e-pre-test-on-child-pipelines/.gitlab-ci.yml#L322-L323

**Review by chouetz — APPROVED** [review]

**Pull request merged**